gear/gear/database.py

```diff
@@ -87,17 +135,20 @@ def __init__(self):
         await aexit(self.conn_context_manager)
         self.conn_context_manager = None
 
+    @retry_coroutine
     async def just_execute(self, sql, args=None):
         assert self.conn
         async with self.conn.cursor() as cursor:
             await cursor.execute(sql, args)
 
+    @retry_coroutine
     async def execute_and_fetchone(self, sql, args=None):
         assert self.conn
         async with self.conn.cursor() as cursor:
             await cursor.execute(sql, args)
             return await cursor.fetchone()
 
+    @retry_async_generator
     async def execute_and_fetchall(self, sql, args=None):
```

Comment:
Instead of retrying here, retry the call to cursor.fetchmany.  You'll need to define an internal function to decorate.  Then I think you can get rid of retry_async_gnerator.